doc/developer/index.adoc

```diff
@@ -26,29 +26,136 @@ These conditions are to ensure that existing user code will continue to work whe
+without the need for a formal review.
+
+Plans to do a patch release are announced by the project lead on the
+https://dev.eclipse.org/mailman/listinfo/rdf4j-dev[rdf4j-dev@eclipse.org mailinglist],
+usually about a week in advance, with an open invitation for contributors to
+propose additional fixes to include, which are done as Pull Requests to the
+_master_ branch.
 
-A patch release can be done in an ad-hoc fashion, without the need for a formal review. Plans to do a patch release are announced by the project lead on the https://dev.eclipse.org/mailman/listinfo/rdf4j-dev[rdf4j-dev@eclipse.org mailinglist], usually about a week in advance, with an open invitation for contributers to propose additional fixes to include.
+=== Creating a patch release branch
 
-=== Creating a release branch
+Any fixes to be included in the a patch release must be merged into the _master_
+branch first.
+A patch release branch should differ from the _master_ branch, at the time of
+release, only by the version number - a patch release branch has a patch number
+version, while the _master_ branch has a SNAPSHOT version.
+To create a patch release branch, follow these steps:
+
+1. Check out the _master_ branch.
++
+E.g. if we're preparing a release 2.2.1, the _master_ branch will have the version 2.2-SNAPSHOT:
++
+    git checkout master
 
-A patch release uses its immediate predecessor release as its basis. This means we need to create a release branch not by simply branching from the current master branch, but by branching from a specific release tag. To create a patch release branch, follow these steps:
+2. Create a new release branch, named `releases/<version>`:
++
+    git checkout -b releases/2.2.1
+
+3. Fix maven version numbers in the release branch.
++
+We need to set the project version to `2.2.1` by using:
++
+    mvn versions:set
++
+This will ask for the new version number. Enter `2.2.1` to indicate that this is the release branch for 2.2.1 release.
++
+After this is done, execute
++
+    mvn versions:commit
++
+which will remove backup files.
++
+Finally, commit the version number changes:
++
+    git commit -s -a -m "release branch for 2.2.1"
+
+4. Tag the version and push the tag upstream:
++
+    git tag 2.2.1
+    git push -u origin 2.2.1
+
+== Hotfix releases
+
+Hotfix release are patch releases that target an prior minor version (not the
+latest stable release). These are needed when a critical bug was found in a
+production deployment using an earlier version.
+
+A Hotfix release use a preceding release as its basis. This means we need to create a release branch not by simply branching from the current _master_ branch, but by branching from a specific release tag. To create a patch release branch, follow these steps:
 
 1. Check out the tag of the previous release.
 +
-E.g. if we're preparing a release 2.1.6, the previous release is 2.1.5, so we do:
+E.g. if we're preparing a release 2.1.6, the preceding release is 2.1.5, so we do:
```

Comment:
Hotfixes are not based on master, they are based on a preceding release (not the latest). Like a bug fix release for java7 users.